app/javascript/packs/userProfileSettings.js

```diff
@@ -1,31 +1,25 @@
-const userFieldIds = ['user[name]', 'user[email]', 'user[username]'];
+const userSettingForm = document.getElementById('user-profile-form');
 const profileFieldIds = Array.from(
   document.querySelectorAll('[id^="profile["]'),
 ).map((node) => node.id);
-const allFieldIds = [...userFieldIds, ...profileFieldIds];
 
 export function fieldCharacterLimits() {
-  window.addEventListener('load', () => {
-    allFieldIds.forEach((field_id) => {
-      const field = document.getElementById(field_id);
-      const fieldValueLength = field.value.length;
-      const fieldCharacterSpan = document.getElementById(
-        field.dataset.characterSpanId,
-      );
-      fieldCharacterSpan.innerHTML = fieldValueLength;
-    });
+  profileFieldIds.forEach((field_id) => {
+    const field = document.getElementById(field_id);
```

Comment:
nitpick (not blocking) - it looks like on line 2, we get all the field elements, and transform that into an array of IDs. And then on this line we get the element by ID. Why not just iterate over the array generated from `document.querySelectorAll('[id^="profile["]')` to save re-finding the field elements?

Reply:
Great point @aitchiss! `document.querySelectorAll('[id^="profile["]')` yields a NodeList, not an array; however, I believe I can iterate over the NodeList using `forEach`. Let me experiment; I may update this code accordingly.